### PR TITLE
applied from_moves

### PR DESCRIPTION
**Pull request by Jamuro-g**
#### Summary
Infrastructure "Applied new from_moves function"

#### Purpose of change
A previous pr of mine added the method: time_duration::from_moves() which turns moves into a turn equivalent time_duration.
Several places in the code did this conversion locally, doing it in one method ensures that changes can be applied easier throughout.

#### Describe the solution
As requested in #47174 i went through the code to check for other places where this conversion was done by hand (moves divided by 100) and turned into/used as a time_duration) and replaced those with the from_moves method

#### Describe alternatives you've considered
I had a bit of an issue with the operator* whenever two time_durations would end up being used (the compiler wasn't sure which of the operator overrides to apply).
The workaround was to use to_turns<int>() to convert the modified time_duration back into an int for the multiplication.

#### Testing
- Played for a few hours, trying to use as many of the areas that i changed code in and kept an eye on the time any activities took.

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -713,7 +713,7 @@ std::string spell::duration_string() const
 
 time_duration spell::duration_turns() const
 {
-    return 1_turns * duration() / 100;
+    return 1_turns * to_turns<int>( time_duration::from_moves( duration() ) );
```
Comment (Qrox) [inline src/magic.cpp:716]: I think `time_duration::from_moves( duration() )` alone is sufficient.

File: src/magic_spell_effect.cpp [hunk 2]
```diff
@@ -428,7 +428,7 @@ static std::set<tripoint> spell_effect_area( const spell &sp, const tripoint &ta
 static void add_effect_to_target( const tripoint &target, const spell &sp )
 {
     const int dur_moves = sp.duration();
-    const time_duration dur_td = 1_turns * dur_moves / 100;
+    const time_duration dur_td = 1_turns * to_turns<int>( time_duration::from_moves( dur_moves ) );
```
Comment (Qrox) [inline src/magic_spell_effect.cpp:431]: Same here.

File: src/vehicle_use.cpp [hunk 3]
```diff
@@ -1032,10 +1032,10 @@ bool vehicle::start_engine( const int e )
                            "engine_single_click_fail" );
             return false;
         }
-        // TODO: start_moves is in moves, but it's an integer, convert it to some time class
+
         const int start_draw_bat = power_to_energy_bat( engine_power *
                                    ( 1.0 + dmg / 2 + cold_factor / 5 ) * 10,
-                                   1_turns * start_moves / 100 );
+                                   1_turns * to_turns<int>( time_duration::from_moves( start_moves ) ) );
```
Comment (Qrox) [inline src/vehicle_use.cpp:1038]: And here.

File: src/magic.cpp (hunk 1 above)
Comment (Jamuro-g) [reply src/magic.cpp:716]: yeah i originally tried that, but sadly the operator overrides are a bit wonky.
I end up with an unambiguous *operator.

There are already two overrides, with each having one of the two params of a multiplication as a time_duration and the other as a cont T.
That in turn means that both would qualify for a multiplication of two time_durations with each other and the compiler realy isn't a fan of that.

Comment (Qrox) [reply src/magic.cpp:716]:
```suggestion
    return time_duration::from_moves( duration() );
```
This is what I mean, the multiplication is not needed.

Comment (Jamuro-g) [reply src/magic.cpp:716]: i think we both mean the same, but just to make 100% sure.

i think it should be return from_turns(duration()) not from_turns(duration()/100))
the reason being that the original division by 100 was followed up by a multiplication by 100 (1_turns)

Comment (Qrox) [reply src/magic.cpp:716]: `from_turns(duration() / 100)` or `from_moves(duration()), because judging from the original code, `duration()` is in moves.

Comment (Jamuro-g) [reply src/magic.cpp]: should be resolved in the updated version

**Pull request merged**